services/subscriber/service.go

```diff
@@ -183,20 +183,28 @@ func (s *Service) createSubscription(se subEntry, mode string, destinations []st
 	default:
 		return nil, fmt.Errorf("unknown balance mode %q", mode)
 	}
-	writers := make([]PointsWriter, len(destinations))
-	stats := make([]writerStats, len(writers))
-	for i, dest := range destinations {
+	writers := []PointsWriter{}
+	stats := []writerStats{}
+	// add only valid destinations
+	for _, dest := range destinations {
 		u, err := url.Parse(dest)
 		if err != nil {
-			return nil, err
+			s.Logger.Println("failed to parse subscription url:", dest)
+			continue
```

Comment:
What's the rationale for ignoring (aside from logging the error) any invalid subscriptions? It's unlikely operators are going to notice that their subscriptions have not been added unless they're checking the logs.